### PR TITLE
Add roadmap document

### PR DESCRIPTION
**Pull request by fgalan**
Fixes https://github.com/telefonicaid/iotagent-node-lib/issues/756

@dcalvoalonso could you have a look and provide LGTM if ok, pls? Thanks!

### CODE REVIEW

File: docs/roadmap.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  out of the scope of GE owners. Thus, the moment in which these features will be
+  finalized cannot be assured.
+  
+### Short term
+
+The following list of features are planned to be addressed in the short term, and incorporated in a release of the product:
+- Removal of warnings and deprecation warnings.
+
+
+### Medium term
```
Comment (dcalvoalonso) [inline docs/roadmap.md]: Deprecation of NGSIv1 could be also addressed in the medium term.

File: docs/roadmap.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  to be done along time which is fully valid only at the time of writing it. This
+  roadmap has not be understood as a commitment on features and/or dates.
+* Some of the roadmap items may be implemented by external community developers, 
+  out of the scope of GE owners. Thus, the moment in which these features will be
+  finalized cannot be assured.
+  
+### Short term
+
+The following list of features are planned to be addressed in the short term, and incorporated in a release of the product:
+- Removal of warnings and deprecation warnings.
```
Comment (dcalvoalonso) [inline docs/roadmap.md]: Fix of bugs reported in GitHub repository, merge of open PRs

Comment (fgalan) [reply docs/roadmap.md]: "Fix of bugs reported in GitHub repository, merge of open PRs"

We receive issue report in a regular basic so I think it is more a support work than a planned item for a roadmap. So in this case, I think is better not include it so NTC.

Thanks anyway for the suggestion!

File: docs/roadmap.md (hunk 1 above)
Comment (fgalan) [reply docs/roadmap.md]: Ok! However, I have adjusted the wording a bit:

"Removal support for NGSIv1 (which currently is deprecated)"

Being strict NGSIv1 is *already* deprecated, so what is pending is to remove the support.

Comment (fgalan) [reply docs/roadmap.md]: Fixed in 70be5fa77d20393f49e521adf56f12754114b4c5

**Pull request merged**